lisa/analysis/tasks.py

```diff
@@ -498,6 +527,33 @@ def plotter(axis, local_fig):
 
         return self.do_plot(plotter, filepath, axis, height=8)
 
+    @df_tasks_total_residency.used_events
+    def plot_tasks_total_residency(self, tasks, ascending=False, filepath=None, axis=None):
+        """
+        Plot the tacked total time spent by each task on each CPU
+
+        :param task: The tasks name or PIDs
+        :type task: list of str or int
+
+        :param ascending: Set True to plot before short running tasks,
+                          False by deafult
+        :type ascending: bool
+
+        .. seealso:: :meth:`lisa.analysis.base.AnalysisHelpers.do_plot`
+        """
+        df = self.df_tasks_total_residency(tasks, ascending)
+
+        def plotter(axis, local_fig):
+            df.T.plot.barh(ax=axis, stacked=True)
+            axis.set_title("Stacked CPU residency of selected tasks")
+            axis.set_ylabel("CPU")
+            axis.set_xlabel("Runtime (s)")
+            axis.grid(True)
+            plt.legend(loc='upper left',ncol=5, bbox_to_anchor=(0, -.15))
```

Comment:
Is `matplotlib` that bad at finding a good location for the legend?

Also, I dislike using `plt` for configuring a graph, because it implicitly operates on the last graph that was generated. AFAICT It's safe to do it in LISA's analysis, but it feels icky. Can't we call `legend` on `axis` instead?

Reply:
It is... and with many tasks, it always ends up in covering some data since it forces the legend within the plot. What I want instead is the legend always _below_ the plot, that's the only way I found to always get a clean plot.

Regarding `plt` vs `axis`... you right, I had the same thought but I forgot to update it.
Using `axis` seems to be working... will update ;)